_posts/2015-11-13-svg-sprites-angular.md

```diff
@@ -0,0 +1,167 @@
+- < IE9 is not supporting svgs
+- Using `<base>` changes behavior in firefox (more below)
+
+Currently, we are embedding svg sprites in our template, just after opening `<body>` tag for good cross browser support. If you want to keep you resources out of your template, you can use a great shim written by Jonathan Neal [svg4everybody] (https://github.com/jonathantneal/svg4everybody).
+
+And if you need to support older browsers (IE8 etc.) you can use image as fallback.
+
+## Examples
+
+[Here] (https://github.com/SinisaG/svg-blog-demo) are a few examples that shows some of the browser glitches.
+
+You can run the examples using [5minutefork] (http://5minfork.com/). More in readme....
+
+## Base tag
+
+If you are like us and use [AngularJs] (https://angularjs.org/) everything written above will apply if you are using urls with hashbang (#!). But if you are using pretty urls with `<base>` tag or you use this tag for any other reason, then be aware, that firefox is interpreting this differently then other browsers, so you will need to reference svgs with an absolute url. For angular or similar environments we suggest using a directive that keeps urls up to date:
+
+
+```javascript
+angular
+    .module('MySvgModule')
+    .directive('ofSvgIcon', ofSvgIcon);
+
+function ofSvgIcon ($location) {
+    return {
+        restrict: 'EA',
+        scope: {
+          icon: '='
+        },
+        link: function (scope, element) {
+
+            // you can modify your url here if needed
+            var getSvgUrl = function (icon) {
+                    return $location.absUrl() + icon;
+            };
+
+            //update xlink:href
+            var setSvgPath = function () {
+                element[0].setAttributeNS("http://www.w3.org/1999/xlink","xlink:href", getSvgPath(scope.icon));
+            };
+
+            // set icons on load
+            setSvgPath();
+
+            // subscribe to update event
+            $rootScope.$on('update.svg', setSvgPath);
+        }
+    };
+}
+
+```
+
+
+Additionally, you have to trigger path update anytime url changes (we use ui.router, event might be called differently if you use different routing). You can also put this straight in to your directive and leave out `update.svg` event, but we think it is good practice to add one more level of abstraction. It makes your code more immune to router changes - if you change the router and event name changes, you have to adjust your code only in one place.
+
+```javascript
+$rootScope.$on('$stateChangeSuccess', function () {
+    $rootScope.$broadcast('update.svg');
+});
+```
+
+Usage (you can also make template part of directive):
+
+```
+<svg class="icon-nav-info">
+    <use of-svg-icon icon="'#icon-nav-admin'"></use>
+</svg>
+```
+
+That has proven to work for us in all browsers we tend to support. Live example - [Onefootball - best app for football] (https://www.onefootball.com) ;)
+
+I hope this clears sprites and svg sprites for you. Happy performance improving!
```

Comment:
Maybe we could clonclude by listing some resources about SVG sprites, like Sara Soueidan's articles https://24ways.org/2014/an-overview-of-svg-sprite-creation-techniques/ & http://sarasoueidan.com/blog/svg-tips-for-designers/, or that SVG sandbox https://github.com/metamn/svg, or talk about other uses of SVG like fonts http://caniuse.com/#feat=svg-fonts